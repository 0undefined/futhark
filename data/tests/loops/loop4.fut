-- Nasty loop whose size cannot be predicted in advance.
-- ==
-- input {
--   [1,2,3]
--   4
-- }
-- output {
--   [1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3]
-- }

fun []int main([]int xs, int n) =
  loop (xs) = for i < n do
    concat(xs,xs)
  in xs
