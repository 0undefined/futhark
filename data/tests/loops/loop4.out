{[1.000000], empty([real])}
